dvc/commands/data.py

```diff
@@ -52,6 +52,7 @@ def _process_status(status: "DataStatus"):
                     file: state
                     for state, files in stage_status.items()
                     for file in files
+                    if state != "unknown"
```

Comment:
Skipping `unknown` state in CLI, `--json` includes this by default.

Reply:
Why? Seems useful to me. Makes more clear what's happening compared to wondering why I asked for `--granular` but only get dir-level output.

Also, this PR seems to blur the line between default behavior and `--with-dirs`. Would it be better to make `--with-dirs` default instead of an option?

Reply:
> Seems useful to me. Makes more clear what's happening compared to wondering why I asked for --granular but only get dir-level output.

I am worried that, it'll dump too much information that is implicitly known - if we don't show granular output, we don't know the output. But I don't have strong opinion here.

> Also, this PR seems to blur the line between default behavior and --with-dirs. Would it be better to make --with-dirs default instead of an option?

What I tried to do there was reduce unnecessary information. Eg: 
```
Dvc Committed changes:
modified: data/
modified: data/foo
```

Usually, `data/` is modified because `data/foo` is modified. So, with `--with-dirs`, we'll show the directories,
but without the flag, it means we'll try our best to show a proper status.
  
 But the implementation has been buggy, that I am also in favour of getting rid of it.

Reply:
> I am worried that, it'll dump too much information that is implicitly known - if we don't show granular output, we don't know the output. But I don't have a strong opinion here.

Will it show any more info than it would if the status was known? I prefer the consistency of showing the same paths whether the status is known or unknown. This is specific to people who pass `--granular` after all, right?

These might be somewhat style/UI guideline opinions, but I think it's better to default to consistent output and worry later if we need to reduce noise.

> But the implementation has been buggy, that I am also in favour of getting rid of it. what do you think?

Yup, again it makes sense to want to reduce the noise, but doesn't feel like it's worth the effort or inconsistency for now.